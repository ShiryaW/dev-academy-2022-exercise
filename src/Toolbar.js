import React from "react";
import PropTypes from "prop-types";
import ButtonGroup from "@mui/material/ButtonGroup";
import Button from "@mui/material/Button";

export const Toolbar = ({ farms, onButtonClick }) => {
  const getButtons = (farms) => {
    if (!farms || farms.length === 0) return;
    return farms.map((farm, idx) => (
      <Button key={idx} onClick={onButtonClick}>
        {farm.name}
      </Button>
    ));
  };

  const farmButtons = getButtons(farms);
  const hasLoaded = farms.length > 0;

  return (
    <div id="toolbar">
      <ButtonGroup disableElevation={true} fullWidth={true} variant="contained">
        {!hasLoaded ? (
          <Button key="loading">Loading farm data...</Button>
        ) : (
          <>
            <Button key="all" onClick={onButtonClick}>
              All
            </Button>
            {farmButtons}
          </>
        )}
      </ButtonGroup>
    </div>
  );
};

Toolbar.propTypes = {
  farms: PropTypes.array.isRequired,
  onButtonClick: PropTypes.func.isRequired,
};
