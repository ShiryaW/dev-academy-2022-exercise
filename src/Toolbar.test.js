import React from "react";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { Toolbar } from "./Toolbar";

configure({ adapter: new Adapter() });

const noOp = jest.fn();
const createToolbar = (props) =>
  shallow(<Toolbar {...{ ...props, onButtonClick: noOp }} />);

describe("Toolbar", () => {
  it("renders the toolbar with toggle buttons for each farm", () => {
    const farms = [
      { farm_id: 1, name: "Friman Metsola" },
      { farm_id: 2, name: "Noora's farm" },
      { farm_id: 3, name: "Organic Ossis's Impact That Lasts" },
      { farm_id: 4, name: "Partial Tech" },
    ];
    const toolbar = createToolbar({ farms });

    expect(toolbar).toMatchSnapshot();
  });

  it("renders the toolbar with loading text if no farms are supplied (have not loaded yet)", () => {
    const toolbar = createToolbar({ farms: [] });
    expect(toolbar).toMatchSnapshot();
  });
});
