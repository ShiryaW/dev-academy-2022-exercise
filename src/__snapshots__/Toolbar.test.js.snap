// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toolbar renders the toolbar with loading text if no farms are supplied (have not loaded yet) 1`] = `
<div
  id="toolbar"
>
  <ForwardRef(ButtonGroup)
    disableElevation={true}
    fullWidth={true}
    variant="contained"
  >
    <ForwardRef(Button)
      disabled={true}
      key="loading"
    >
      Loading farm data...
    </ForwardRef(Button)>
  </ForwardRef(ButtonGroup)>
</div>
`;

exports[`Toolbar renders the toolbar with toggle buttons for each farm 1`] = `
<div
  id="toolbar"
>
  <ForwardRef(ButtonGroup)
    disableElevation={true}
    fullWidth={true}
    variant="contained"
  >
    <ForwardRef(Button)
      key="all"
      onClick={[MockFunction]}
    >
      All
    </ForwardRef(Button)>
    <ForwardRef(Button)
      key="0"
      onClick={[MockFunction]}
    >
      Friman Metsola
    </ForwardRef(Button)>
    <ForwardRef(Button)
      key="1"
      onClick={[MockFunction]}
    >
      Noora's farm
    </ForwardRef(Button)>
    <ForwardRef(Button)
      key="2"
      onClick={[MockFunction]}
    >
      Organic Ossis's Impact That Lasts
    </ForwardRef(Button)>
    <ForwardRef(Button)
      key="3"
      onClick={[MockFunction]}
    >
      Partial Tech
    </ForwardRef(Button)>
  </ForwardRef(ButtonGroup)>
</div>
`;
